### PR TITLE
[18] Update ReactDOMClient docs

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

Some changes/mistakes since the last RC.

### CODE REVIEW

**Review by salazarm — APPROVED** [review]
good catch by seb

File: content/docs/reference-react-dom.md [hunk 1]
```diff
@@ -85,14 +85,14 @@ Same as [`createRoot()`](#createroot), but is used to hydrate a container whose
 ### `createPortal()` {#createportal}
 
 ```javascript
-ReactDOM.createPortal(child, container)
+ReactDOMClient.createPortal(child, container)
```
Comment (sebmarkbage) [inline content/docs/reference-react-dom.md:88]: All of these are *not* on ReactDOMClient. Only hydrateRoot and createRoot are.

File: content/docs/reference-react-dom.md [hunk 2]
```diff
@@ -55,25 +55,25 @@ root.unmount();
...
+ReactDOMClient.hydrateRoot(element, container[, options])
 ```
 
 Same as [`createRoot()`](#createroot), but is used to hydrate a container whose HTML contents were rendered by [`ReactDOMServer`](/docs/react-dom-server.html). React will attempt to attach event listeners to the existing markup.
 
-`hydrateRoot` accepts two optional callbacks as options:
-- `onDeleted`: callback called when content is deleted.
-- `onHydrated`: callback called after hydration completes.
+`hydrateRoot` accepts two options:
+- `onRecoverableError`: optional callback called when React automatically recovers from errors during hydration.
```
Comment (sebmarkbage) [inline content/docs/reference-react-dom.md:75]: This is not just during hydration. It's also for concurrency issues.

File: content/docs/reference-react-dom.md [hunk 3]
```diff
@@ -55,25 +55,25 @@ root.unmount();
...
-ReactDOM.hydrateRoot(element, container[, options])
+ReactDOMClient.hydrateRoot(element, container[, options])
 ```
 
 Same as [`createRoot()`](#createroot), but is used to hydrate a container whose HTML contents were rendered by [`ReactDOMServer`](/docs/react-dom-server.html). React will attempt to attach event listeners to the existing markup.
 
-`hydrateRoot` accepts two optional callbacks as options:
-- `onDeleted`: callback called when content is deleted.
-- `onHydrated`: callback called after hydration completes.
+`hydrateRoot` accepts two options:
```
Comment (sebmarkbage) [inline content/docs/reference-react-dom.md:74]: These options can also be passed createRoot.

**Pull request merged**